org.opentosca.container.core.impl/src/org/opentosca/container/core/impl/plan/PlanInvocationEngine.java

```diff
@@ -502,7 +507,15 @@ public void invokePlan(final CSARID csarID, final QName serviceTemplateId, final
         }
     }
 
-
+    private Map<String,String> transform(List<TParameterDTO> params){
+        Map<String,String> result = new HashMap<String,String>();
+        
+        for(TParameterDTO param: params) {
+            result.put(param.getName(), param.getValue());
+        }
+        
+        return result;
```

Comment:
Could be reframed using `Collectors.toMap(TParameterDTO::getName, TParameterDTO::getValue)`

Reply:
Most people won't get it, but done